rust/lance-namespace/src/rest.rs

```diff
@@ -0,0 +1,547 @@
+        
+        let client = client_builder.build()?;
+        
+        Ok(RestNamespace {
+            client,
+            base_url,
+            config,
+        })
+    }
+    
+    fn object_id_str(&self, id: &[String]) -> String {
+        if id.is_empty() {
+            self.config.delimiter().to_string()
+        } else {
+            id.join(&self.config.delimiter())
+        }
+    }
+    
+    async fn make_request<T>(&self, method: reqwest::Method, path: &str, body: Option<Vec<u8>>) -> Result<T>
+    where
+        T: serde::de::DeserializeOwned,
+    {
+        let url = format!("{}/{}", self.base_url.trim_end_matches('/'), path.trim_start_matches('/'));
+        
+        let mut request = self.client.request(method, &url);
+        
+        if let Some(body) = body {
+            request = request.body(body).header("Content-Type", "application/json");
+        }
+        
+        let response = request.send().await?;
+        
+        if !response.status().is_success() {
+            return Err(NamespaceError::Runtime(format!(
+                "HTTP request failed with status {}: {}",
+                response.status(),
+                response.text().await.unwrap_or_default()
+            )));
+        }
+        
+        let response_body = response.bytes().await?;
+        
+        if response_body.is_empty() {
+            // Handle empty response for operations that don't return data
+            return serde_json::from_str("{}").map_err(Into::into);
+        }
+        
+        serde_json::from_slice(&response_body).map_err(Into::into)
+    }
+}
+
+#[async_trait]
+impl LanceNamespace for RestNamespace {
+    async fn list_namespaces(&self, request: ListNamespacesRequest) -> Result<ListNamespacesResponse> {
+        let id_str = self.object_id_str(request.id.as_deref().unwrap_or_default());
+        
+        let mut path = format!("namespaces/{}", id_str);
+        let mut query_params = Vec::new();
+        
+        query_params.push(format!("delimiter={}", urlencoding::encode(&self.config.delimiter())));
+        
+        if let Some(page_token) = &request.page_token {
+            query_params.push(format!("page_token={}", urlencoding::encode(page_token)));
+        }
+        
+        if let Some(limit) = request.limit {
+            query_params.push(format!("limit={}", limit));
+        }
+        
+        if !query_params.is_empty() {
+            path.push('?');
+            path.push_str(&query_params.join("&"));
+        }
+        
+        self.make_request(reqwest::Method::GET, &path, None).await
+    }
+    
+    async fn describe_namespace(&self, request: DescribeNamespaceRequest) -> Result<DescribeNamespaceResponse> {
+        let id_str = self.object_id_str(request.id.as_deref().unwrap_or_default());
+        
+        let path = format!("namespaces/{}?delimiter={}", id_str, urlencoding::encode(&self.config.delimiter()));
```

Comment:
we should have a shared function to handle query parameter all at once. For example, we can consult using something similar to https://github.com/lancedb/lancedb/blob/main/rust/lancedb/src/remote/client.rs